pkg/build/api/types.go

```diff
@@ -475,6 +475,10 @@ const (
 	// build pod but one was already present.
 	StatusReasonBuildPodExists StatusReason = "BuildPodExists"
 
+	// StatusReasonInvalidEnvironmentVariable indicates that an error occurred processing
+	// the supplied options for environment variables in the build strategy environment
+	StatusReasonInvalidEnvironmentVariable StatusReason = "InvalidEnvironmentVariable"
```

Comment:
There's a bit of a mismatch between the name of this error and the actual reasons why `ProcessValueFrom` might return - is it possible to tighten this up at all?  How do other parts of origin/k8s report this type of error?